Assets/Scripts/Generic Event Invokers/GearLever.cs

```diff
@@ -0,0 +1,50 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class GearLever : GenericInteractionToggle
+{
+    [SerializeField]
+    private List<GameObject> gears = new();
+
+    readonly private List<IMustBeWaited> itemsToWaitFor = new();
+
+    private void Awake()
+    {
+        foreach (var go in gears)
+        {
+            itemsToWaitFor.Add(go.GetComponent<IMustBeWaited>());
+
+            //me daba paja agregar uno por uno en el inspector para las palancas, verdaderamente es la unica razon por la que vive esta clase
+            var gear = go.GetComponent<Gear>();
```

Comment:
si pones este var gear arriba del itemsto wait for podes poner  itemsToWaitFor.Add(gear) y te ahorras un GetComponent

Reply:
ahh mall